securesignals-ima-devapp/src/main/java/com/uid2/dev/MainActivity.java

```diff
@@ -0,0 +1,283 @@
+package com.uid2.dev;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.MediaController;
+import android.widget.VideoView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.ads.interactivemedia.v3.api.AdDisplayContainer;
+import com.google.ads.interactivemedia.v3.api.AdErrorEvent;
+import com.google.ads.interactivemedia.v3.api.AdEvent;
+import com.google.ads.interactivemedia.v3.api.AdsLoader;
+import com.google.ads.interactivemedia.v3.api.AdsManager;
+import com.google.ads.interactivemedia.v3.api.AdsManagerLoadedEvent;
+import com.google.ads.interactivemedia.v3.api.AdsRenderingSettings;
+import com.google.ads.interactivemedia.v3.api.AdsRequest;
+import com.google.ads.interactivemedia.v3.api.ImaSdkFactory;
+import com.google.ads.interactivemedia.v3.api.ImaSdkSettings;
+import com.google.ads.interactivemedia.v3.api.player.VideoProgressUpdate;
+import com.uid2.UID2Manager;
+import com.uid2.data.UID2Identity;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+
+/** Main activity. */
+public class MainActivity extends AppCompatActivity {
+
+    private static final String LOGTAG = "IMABasicSample";
+    private static final String SAMPLE_VIDEO_URL =
+        "https://storage.googleapis.com/gvabox/media/samples/stock.mp4";
+
+    /**
+     * IMA sample tag for a single skippable inline video ad. See more IMA sample tags at
+     * https://developers.google.com/interactive-media-ads/docs/sdks/html5/client-side/tags
+     */
+    private static final String SAMPLE_VAST_TAG_URL =
+        "https://pubads.g.doubleclick.net/gampad/ads?iu=/21775744923/external/"
+            + "single_preroll_skippable&sz=640x480&ciu_szs=300x250%2C728x90&gdfp_req=1&output=vast"
+            + "&unviewed_position_start=1&env=vp&impl=s&correlator=";
+
+    // Factory class for creating SDK objects.
+    private ImaSdkFactory sdkFactory;
+
+    // The AdsLoader instance exposes the requestAds method.
+    private AdsLoader adsLoader;
+
+    // AdsManager exposes methods to control ad playback and listen to ad events.
+    private AdsManager adsManager;
+
+    // The saved content position, used to resumed content following an ad break.
+    private int savedPosition = 0;
+
+    // This sample uses a VideoView for content and ad playback. For production
+    // apps, Android's Exoplayer offers a more fully featured player compared to
```

Comment:
Nit: I know this is copied from a different app, but they are using a different max line length than we have configured for this project. Consider reformatting these comment blocks to use all `120` characters.